src/ape/managers/chain.py

```diff
@@ -1075,3 +1092,12 @@ def set_balance(self, account: Union[BaseAddress, AddressType], amount: Union[in
             amount = int(amount, 16)
 
         return self.provider.set_balance(account, amount)
+
+    def get_receipt(self, transaction_hash: str) -> ReceiptAPI:
+        cached_receipt = self.chain_manager.account_history.get_receipt(transaction_hash)
+        if cached_receipt:
+            return cached_receipt
+
+        receipt = self.provider.get_receipt(transaction_hash)
+        self.chain_manager.account_history.append(receipt)
```

Comment:
Once the query manager gives us receipts instead of transactions, we can use the query manager here.

Reply:
One of the things I wanted to say